### PR TITLE
Replace Item.hidden_if_available with relationship to other Item

### PR DESCRIPTION
**Pull request by raphaelm**
- [x] Implementation
- [x] Tests

### CODE REVIEW

File: src/pretix/base/models/items.py [hunk 1]
```diff
@@ -501,6 +501,17 @@ class Item(LoggedModel):
                     "in which both products are visible while all tickets in the referenced quota are reserved, "
                     "but not yet sold.")
     )
+    hidden_if_item_available = models.ForeignKey(
+        'Item',
+        null=True, blank=True,
+        on_delete=models.SET_NULL,
+        verbose_name=_("Only show after sellout of"),
```
Comment (wiffbi) [inline src/pretix/base/models/items.py:508]: For old events using the quota-setting at least the german translation fails when using the same string in english. Not sure it is worth the effort though as `hidden_if_available` is anyways deprecated.

File: src/pretix/locale/de/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -4255,7 +4255,7 @@ msgstr "Produktbild"
 
 #: pretix/base/models/items.py:491
 msgid "Only show after sellout of"
-msgstr "Nicht anzeigen, wenn anderes Kontingent verfügbar"
+msgstr "Nicht anzeigen, wenn anderes Produkt verfügbar"
```
Comment (wiffbi) [inline src/pretix/locale/de/LC_MESSAGES/django.po:4258]: See comment above. Maybe a more generic translation helps?

```suggestion
msgstr "Nicht anzeigen, wenn verfügbar"
```

File: src/pretix/locale/de_Informal/LC_MESSAGES/django.po [hunk 3]
```diff
@@ -4252,7 +4252,7 @@ msgstr "Produktbild"
 
 #: pretix/base/models/items.py:491
 msgid "Only show after sellout of"
-msgstr "Nicht anzeigen, wenn anderes Kontingent verfügbar"
+msgstr "Nicht anzeigen, wenn anderes Produkt verfügbar"
```
Comment (wiffbi) [inline src/pretix/locale/de_Informal/LC_MESSAGES/django.po:4255]: See comment above. Maybe a more generic translation helps?

```suggestion
msgstr "Nicht anzeigen, wenn verfügbar"
```

**Review by wiffbi — COMMENTED** [review]
Looks good, just some minor remarks regarding self-referencing and translation. And a question regarding manually deactivated products.

File: src/pretix/presale/views/event.py [hunk 4]
```diff
@@ -256,6 +281,19 @@ def get_grouped_items(event, subevent=None, voucher=None, channel='web', require
                 item._remove = True
                 continue
 
+        if item.hidden_if_item_available:
```
[the remark this answers is not in the document]
Comment (raphaelm) [reply src/pretix/presale/views/event.py:284]: We discussed this in Slack and my impression was that *your* point was that *only* quota should be considered?

File: src/pretix/presale/views/event.py [hunk 5]
```diff
@@ -403,7 +441,7 @@ def get_grouped_items(event, subevent=None, voucher=None, channel='web', require
     if not quota_cache_existed and not voucher and not allow_addons and not base_qs_set and not filter_items and not filter_categories:
         event.cache.set(quota_cache_key, quota_cache, 5)
     items = [item for item in items
-             if (len(item.available_variations) > 0 or not item.has_variations) and not item._remove]
+             if (len(item.available_variations) > 0 or not item.has_variations) and not item._remove and not getattr(item, '_hide_for_available', False)]
```
[the remark this answers is not in the document]
Comment (raphaelm) [reply src/pretix/presale/views/event.py:444]: Yes, will remove

File: src/pretix/presale/views/event.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (raphaelm) [reply src/pretix/presale/views/event.py:284]: In other words: If we consider `active`, why not `available_until`? Why not a transitive `hidden_if_item_available`? Where does it end? ;)

File: src/pretix/base/models/items.py (hunk 1 above)
Comment (raphaelm) [reply src/pretix/base/models/items.py:508]: I don't like the generic German wording and I don't have a better one, so I've added a context parameter

File: src/pretix/presale/views/event.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (wiffbi) [reply src/pretix/presale/views/event.py:284]: You are right, sorry for my confusion!

**Pull request merged**